src/Uno.UI/UI/Xaml/UIElement.Pointers.Skia.cs

```diff
@@ -117,109 +153,99 @@ private void CoreWindow_PointerMoved(CoreWindow sender, PointerEventArgs args)
-					}
-				}
-			}
-
 			private void PropagateEvent(PointerEventArgs args, Action<UIElement> raiseEvent)
 			{
-				if(Window.Current.Content is UIElement root)
+				if(Window.Current.RootElement is UIElement root)
 				{
-					PropagageEventRecursive(args, new Point(0, 0), root, raiseEvent);
+					PropagageEventRecursive(args, new Point(0, 0), root, Matrix3x2.Identity, raiseEvent);
 				}
 			}
 
-			private bool PropagageEventRecursive(PointerEventArgs args, Point root, UIElement element, Action<UIElement> raiseEvent)
+			private bool PropagageEventRecursive(PointerEventArgs args, Point root, UIElement element, Matrix3x2 currentTransform, Action<UIElement> raiseEvent)
 			{
 				bool raised = false;
-				var rect = element.LayoutSlotWithMarginsAndAlignments;
-				rect.X += root.X;
-				rect.Y += root.Y;
+				var elementRect = element.LayoutSlotWithMarginsAndAlignments;
+				elementRect.X += root.X;
+				elementRect.Y += root.Y;
 
 				var position = args.CurrentPoint.Position;
+				var pointer = new Pointer(args.CurrentPoint.PointerId, PointerDeviceType.Mouse, false, isInRange: true);
+				var pointerArgs = new PointerRoutedEventArgs(args, pointer, element);
 
 				if (element.RenderTransform != null)
 				{
-					position = element.RenderTransform.Inverse.TransformPoint(position);
+					currentTransform *= element.RenderTransform.MatrixCore;
+					elementRect = currentTransform.Transform(elementRect);
 				}
 
-				if (position.X >= rect.X	
-					&& position.Y >= rect.Y
-					&& position.X <= rect.X + rect.Width
-					&& position.Y <= rect.Y + rect.Height)
+				if (elementRect.Contains(position))
 				{
-
-					foreach (var e in element.GetChildren().ToArray())
+					foreach (var e in element.GetChildren().Reverse().ToArray())
 					{
-						raised |= PropagageEventRecursive(args, rect.Location, e, raiseEvent);
+						if(PropagageEventRecursive(args, elementRect.Location, e, currentTransform, raiseEvent))
+						{
+							return true;
+						}
 					}
 
 					var isHitTestVisible =
 						element.GetValue(HitTestVisibilityProperty) is HitTestVisibility hitTestVisibility
 						&& hitTestVisibility == HitTestVisibility.Visible;
 
-					if (!raised && isHitTestVisible)
+					if (isHitTestVisible)
 					{
-						if (!element._pointerEntered)
+						if (!element.IsOver(pointer))
 						{
-							// Console.WriteLine($"PointerManager.Entered [{element}/{element.GetHashCode():X8}");
-							element._pointerEntered = true;
-
-							var pointer = new Pointer(args.CurrentPoint.PointerId, PointerDeviceType.Mouse, false, isInRange: true);
-
-							var pointerArgs = new PointerRoutedEventArgs(args, pointer, element);
-							TraverseAncestors(element, pointerArgs, e => e.OnNativePointerEnter(pointerArgs));
+							element.OnNativePointerEnter(pointerArgs);
 						}
 
 						raiseEvent(element);
-						raised = true;
+						return true;
 					}
 				}
 				else
 				{
-					if (element._pointerEntered)
+					bool RecursePointerExited(UIElement e)
```

Comment:
Codacy found an issue: [Change return type to 'void'; not a single caller uses the returned value.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5924508)